server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -635,20 +627,33 @@ public VirtConsumerMap getHostConsumersMap(Owner owner,
         if (hypervisorIds == null || hypervisorIds.isEmpty()) {
             return new LinkedList<Consumer>();
         }
-        return currentSession().createCriteria(Consumer.class)
-            .createAlias("owner", "o")
-            .add(Restrictions.eq("o.key", ownerKey))
-            .createAlias("hypervisorId", "hvsr")
-            .add(getHypervisorIdRestriction(hypervisorIds))
-            .list();
-    }
-
-    private Criterion getHypervisorIdRestriction(Collection<String> hypervisorIds) {
-        List<Criterion> ors = new LinkedList<Criterion>();
+        LinkedList<String> hypervisorIdsUpper = new LinkedList<String>();
+        // Check for hypervisor ids case-insensitively
         for (String hypervisorId : hypervisorIds) {
-            ors.add(Restrictions.eq("hvsr.hypervisorId", hypervisorId).ignoreCase());
+            hypervisorIdsUpper.add(hypervisorId.toUpperCase());
+        }
+        List<Consumer> consumers = new LinkedList<Consumer>();
+        String query = "SELECT c FROM Consumer c WHERE " +
```

Comment:
Please add some descriptive comments here for history when someone is browsing the code later.